### DISCUSSION TITLE
How to check if package is installed in Node script?

### DISCUSSION THREAD

**Discussion by andyslack** [pnpm/pnpm#7266] [Q&A] [answered]
Hi Guys,

We have recently migrated from NPM to PNPM and everything is working well except one peice of code.

We have a function which checks to see if a package is installed, this happens before we lazy load the module. 

We do this to check which "plugins" the client has installed.

The code looks like this:

```
export class Modules {
	static isInstalled(name: string): boolean {
		try {
			const p = require.resolve(name)
			return !!p
		} catch (e: any) {
			const logger = new Logger()
			logger.debug(`[utils::modules::isInstalled] ${e.message}`, e)
			return false
		}
	}
}
```

Since migrating to PNPM this no longer works, which is breaking a number of our packages.

What is the best way to fix this?

Thanks in advanced.

**Comment by zkochan** [comment, marked as answer]
Are these plugins installed as direct dependencies? npm hoists everything to the root of node_modules, so this will be able to resolve packages from subdependencies. However, by default pnpm only puts direct dependencies to the root of node_modules. If you want npm's node_modules structure, use the `node-linker=hoisted` setting.

**Comment by andyslack** [comment]
Thats working perfectly, thank you very much @zkochan
